pkg/resource/table/hooks.go

```diff
@@ -444,3 +452,37 @@ func customPreCompare(
 		}
 	}
 }
+
+func equalAttributeDefinitions(a []*v1alpha1.AttributeDefinition, b []*v1alpha1.AttributeDefinition) bool {
+	added, updated, removed := computeAttributeDefinitions(a, b)
+	return len(added) == 0 && len(updated) == 0 && len(removed) == 0
+}
+
+// computeAttributeDefinitions compares two AttributeDefinition arrays and return three different list
+// containing the added, updated and removed tags.
+// The removed tags only contains the Key of tags
+func computeAttributeDefinitions(
```

Comment:
@jaypipes    thanks for comments.
I made it too complex,  now I just copy `qualAttributeDefinitions` from PR #30  to make the resource can be synced after creation.  It would unblock our migration. 😄